http/src/main/resources/init-resources/gce-init.sh

```diff
@@ -60,130 +60,142 @@ display_time() {
+PROXY_SITE_CONF=$(proxySiteConf)
+JUPYTER_SERVER_EXTENSIONS=$(jupyterServerExtensions)
+JUPYTER_NB_EXTENSIONS=$(jupyterNbExtensions)
+JUPYTER_COMBINED_EXTENSIONS=$(jupyterCombinedExtensions)
+JUPYTER_LAB_EXTENSIONS=$(jupyterLabExtensions)
+JUPYTER_USER_SCRIPT_URI=$(jupyterUserScriptUri)
+JUPYTER_USER_SCRIPT_OUTPUT_URI=$(jupyterUserScriptOutputUri)
+JUPYTER_NOTEBOOK_CONFIG_URI=$(jupyterNotebookConfigUri)
+JUPYTER_NOTEBOOK_FRONTEND_CONFIG_URI=$(jupyterNotebookFrontendConfigUri)
+CUSTOM_ENV_VARS_CONFIG_URI=$(customEnvVarsConfigUri)
+RSTUDIO_LICENSE_FILE=$(rstudioLicenseFile)
+
+mkdir -p /work
+mkdir -p /certs
+
+log 'Formatting and mounting persistent disk...'
+
+# Format and mount persistent disk
+export DISK_DEVICE_ID=$(lsblk -o name,serial | grep 'user-disk' | awk '{print $1}')
+# Only format disk is it hasn't already been formatted
+if [ "$IS_GCE_FORMATTED" == "false" ] ; then
+  mkfs.ext4 -m 0 -E lazy_itable_init=0,lazy_journal_init=0,discard /dev/${DISK_DEVICE_ID}
+fi
+mount -o discard,defaults /dev/${DISK_DEVICE_ID} /work
+# Directory to store user installed packages so they persist
+mkdir -p /work/packages
+chmod a+rwx /work/packages
+# Ensure persistent disk re-mounts if runtime stops and restarts
+cp /etc/fstab /etc/fstab.backup
+echo UUID=`blkid -s UUID -o value /dev/${DISK_DEVICE_ID}` /work ext4 discard,defaults,nofail 0 2 | tee -a /etc/fstab
+chmod a+rwx /work
+
+# done persistent disk setup
+STEP_TIMINGS+=($(date +%s))
+
+log 'Copying secrets from GCS...'
+
+# Add the certificates from the bucket to the VM. They are used by the docker-compose file
+gsutil cp ${SERVER_CRT} /certs
+gsutil cp ${SERVER_KEY} /certs
+gsutil cp ${ROOT_CA} /certs
+gsutil cp ${PROXY_SITE_CONF} /etc
+gsutil cp ${JUPYTER_DOCKER_COMPOSE_GCE} /etc
+gsutil cp ${RSTUDIO_DOCKER_COMPOSE} /etc
+gsutil cp ${PROXY_DOCKER_COMPOSE} /etc
+gsutil cp ${WELDER_DOCKER_COMPOSE} /etc
+
+echo "" > /etc/google_application_credentials.env
+
+# Install env var config
+if [ ! -z "$CUSTOM_ENV_VARS_CONFIG_URI" ] ; then
+  log 'Copy custom env vars config...'
+  gsutil cp ${CUSTOM_ENV_VARS_CONFIG_URI} /etc
+fi
 
-#####################################################################################################
-# Set up that IS specific to RUNTIME_OPERATION:
-#
-# We perform some of the operations based on whether we are 'creating' a GCE VM or 'restarting'
-# a VM that was previously created and stopped.
-#####################################################################################################
+# Install RStudio license file, if specified
+if [ ! -z "$RSTUDIO_DOCKER_IMAGE" ] ; then
+  STAT_EXIT_CODE=0
+  gsutil -q stat ${RSTUDIO_LICENSE_FILE} || STAT_EXIT_CODE=$?
+  if [ $STAT_EXIT_CODE -eq 0 ] ; then
+    echo "Using RStudio license file $RSTUDIO_LICENSE_FILE"
+    gsutil cp ${RSTUDIO_LICENSE_FILE} /etc/rstudio-license-file.lic
+  else
+    echo "" > /etc/rstudio-license-file.lic
+  fi
+fi
 
-if [[ "$RUNTIME_OPERATION" == 'creating' ]]; then
```

Comment:
We have separate scripts create VM creation and VM startup, so this `$RUNTIME_OPERATION` variable is never actually used. I removed it from Leo scala code and this script. 

Unfortunately this introduces a lot of whitespace changes to `gce-init.sh`..